src/main/java/fr/openent/viescolaire/service/impl/DefaultInitService.java

```diff
@@ -171,5 +176,46 @@ public Future<JsonObject> initSubjects(String structureId, String label, String
         return promise.future();
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public Future<JsonObject> initServices(String structureId, SubjectModel subject) {
+        Promise<JsonObject> promise = Promise.promise();
+        this.userService.getTeachersWithClassGroupIds(structureId)
+                .onFailure(fail -> {
+                    LOGGER.error(String.format("[Viescolaire@%s::initServices] Failed to retrieve teachers with classes/groups",
+                            this.getClass().getSimpleName()), fail);
+                    promise.fail(fail);
+                })
+                .onSuccess(teachers -> {
+                    List<Future<JsonObject>> createServiceFutures = new ArrayList<>();
+
+                    ((List<JsonObject>) teachers.getList()).forEach(teacher -> {
+                       String teacherId = teacher.getString(Field.ID);
+                       List<String> classIds = teacher.getJsonArray(Field.CLASSIDS, new JsonArray()).getList();
+                       List<String> groupIds = teacher.getJsonArray(Field.GROUPIDS, new JsonArray()).getList();
+                       classIds.addAll(groupIds);
+
+                       if (!classIds.isEmpty()) {
+                           InitServiceModel service = new InitServiceModel(
+                                   new JsonObject()
+                                           .put(Field.ID_MATIERE, subject.getId())
+                                           .put(Field.ID_GROUPES, classIds)
+                                           .put(Field.ID_ENSEIGNANT, teacherId)
+                                           .put(Field.ID_ETABLISSEMENT, structureId));
+
+                           createServiceFutures.add(this.servicesService.createService(service));
+                       }
+                    });
+
+                    FutureHelper.all(createServiceFutures)
+                            .onFailure(fail -> {
+                                LOGGER.error(String.format("[Viescolaire@%s::initServices] Failed to create services",
+                                        this.getClass().getSimpleName()), fail);
+                                promise.fail(fail);
```

Comment:
Le 2eme param ne sera pas pris en compte dans les logs
je te suggère un rajout de %s avec le fail.getMessage()

```suggestion
  LOGGER.error(String.format("[Viescolaire@%s::initServices] Failed to create services: %s",
                                        this.getClass().getSimpleName(), fail.getMessage()));
                                promise.fail(fail.getMessage());
```

de plus, il vaut mieux faire un promise.fail en mettant la trace exception